src/Notifier.js

```diff
@@ -133,29 +153,35 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
+            // Case when we do not have the permission as 'granted'
+            if (this.isPermissionDefault()) {
+                // Attempt to get permission from user
+                var _this = this;
+                global.Notification.requestPermission().then(function(result) {
+                    if (result === 'denied') {
                         dis.dispatch({
                             action: "notifier_enabled",
-                            value: true
+                            value: false
                         });
+                        _this.setToolbarHidden(true);
```

Comment:
If the "You are not receiving desktop notifications" bar disappears, it gives the impression that now you will be receiving desktop notifications.

You're right that clicking on the "Enable them now" link won't work, but maybe we should deal with that a different way. (We could hide 'Enable them now', or show a modal dialog giving the user tips on how to enable notifications). Why don't you leave the toolbar shown as it is for now, and we can consider whether to handle this better in a separate PR.